src/Halogen/HTML/Events.purs

```diff
@@ -100,6 +108,17 @@ onScroll = handler (EventType "scroll")
 onChange :: forall r i. (Event -> Maybe i) -> IProp (onChange :: Event | r) i
 onChange = handler ET.change
 
+onFileUpload :: forall r i t. Unfoldable t => Monoid (t File)
+             => (Effect (t File) -> Maybe i) -> IProp (onFileUpload :: Event | r) i
+onFileUpload f = handler ET.change $
+  ( Event.target >=>
+    HTMLInputElement.fromEventTarget >=>
+    ((HTMLInputElement.files >=> (maybe mempty items >>> pure)) >>> pure) ) >=>
```

Comment:
As you have Unfoldable f constraint you don't need Monoid instance because mempty like value can be constructed like this `unfoldr identity Nothing`. i.e. no need for Monoid constraint.

Reply:
I cannot get `unfoldr identity Nothing`

```shell
An infinite type was inferred for an expression:
Maybe (Tuple t1 t0)
```

`unfoldr (const Nothing) Nothing` seems to work though